File: Homeworking/diamond.py
def diamond(n):
        if n <= 0:
            return(None)
        if n % 2 == 0:
            return("NULL")
        else:
            n = int(n / 2 + 1)
            c = ''
            for x in list(range(n)) + list(reversed(range(n - 1))):
                c += (('{: <{w1}}{:*<{w2}}'.format('', '', w1=n-x-1, w2=x*2+1))+ '\n')
            return(c)
print(diamond(0))

File: Homeworking/diamond.js

function diamond(n){
  var diam = "";
  if (n % 2 === 0 || n < 1) {
    return null;
  }
  else {
  var i, j, k, x = n;
    for (i = 1; i <= x; i+=2) {

      for (j = 0; j <(x-i/2-x/2); j++) {
        diam += " ";
      }

      for (k = 1; k <= i; k++) {
        diam += "*"
      }
      diam += "\n";

    }

    for (i=x-2; i >= 0; i -= 2) {

      for (j = 0; j < (x-i/2-x/2); j++) {
        diam += " ";
      }

      for (k = 1; k <= i; k++) {
        diam += "*"
      }
      diam += "\n";
    }
  }
  return diam;
}
console.log(diamond(5));
